tinygrad/runtime/ops_metal.py

```diff
@@ -72,7 +84,7 @@ def __call__(self, *bufs, global_size:Tuple[int,int,int], local_size:Tuple[int,i
     encoder = command_buffer.computeCommandEncoder()
     encoder.setComputePipelineState_(self.pipeline_state)
     for i,a in enumerate(bufs):
-      if isinstance(a, RawMetalBuffer): encoder.setBuffer_offset_atIndex_(a._buf, 0, i)
+      if isinstance(a, RawMetalBuffer): encoder.setBuffer_offset_atIndex_(a.try_from_file()._buf if i > 0 else a._buf, 0, i) # out buffer at index 0 is never a file
```

Comment:
the issue here is that the out buffer at index 0 has `from_file` set because of the way the AST is but we want to ignore it since we're not writing back to disk in this case